scripts/preprocessing/kfold_partition_dataset.py

```diff
@@ -0,0 +1,137 @@
+""" usage: kfold_partition_dataset.py [-h] [-i IMAGEDIR] [-o OUTPUTDIR] [-k KFOLDS] [-x] [-s SEED]
+
+Partition dataset of images into training and testing sets
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -i IMAGEDIR, --imageDir IMAGEDIR
+                        Path to the folder where the image dataset is stored. If not specified, the CWD will be used.
+  -o OUTPUTDIR, --outputDir OUTPUTDIR
+                        Path to the output folder where the train and test dirs should be created. Defaults to the same directory as IMAGEDIR.
+  -k KFOLDS, --kfolds KFOLDS
+                        The number of folds over the total number of images, used for k-fold cross validation. The default is 10.
+  -x, --xml             Set this flag if you want the xml annotation files to be processed and copied over.
+  -s, --seed SEED       Set the seed for shuffling randomization.
+"""
+import os
+import re
+from shutil import copyfile
+import argparse
+import math
+import random
+
+# Imports for k-fold cross-validation
+from sklearn.model_selection import KFold
+import numpy as np
+
+def iterate_dir(source, dest, k, copy_xml):
+    source = source.replace('\\', '/')
+    dest = dest.replace('\\', '/')
+
+
+    # Shuffle images first so we get a good test.
+    images = [f for f in os.listdir(source)
+              if re.search(r'([a-zA-Z0-9\s_\\.\-\(\):])+(.JPG|.jpeg|.png)$', f)]
+    random.shuffle(images)
+
+    if images == None or len(images) == 0:
+      print("Error: No images found.")
+      exit(1)
+
+    # Fold data set into [k] folds.
+    # For each fold, use 1/[k] as validation data and [k]-1/[k] as test data.
+    np_images = np.array(images)
+    kf = KFold(n_splits=k, shuffle=False)
+    print(kf)
+
+    # Count keeps track of which fold we are in.
+    kfold_count = 0
+    for train_index, test_index in kf.split(np_images):
+      print("Current fold: {}".format(kfold_count))
```

Comment:
random thing, I've found using the f string thing to be faster (in case you didn't know about it): like: `f"Current fold: {kfld_count}`

Reply:
Oh true, thanks. I don't really know the best way to do python string formatting...